legend-pure-code-compiled-core/src/main/resources/core/pure/corefunctions/metaExtension.pure

```diff
@@ -309,23 +309,52 @@ function meta::pure::functions::meta::reverseProperties(class:Class<Any>[1]):Pro
    $t->evaluateAndDeactivate()->filter(p|$p->genericType().rawType == Property)->cast(@Property<Nil,Any|*>);
 }
 
+function meta::pure::functions::meta::shouldRecurse(fd:FunctionDefinition<Any>[1], func:Function<{FunctionDefinition<Any>[1]->Boolean[1]}>[1]):Boolean[1]
+{
+   $func->eval($fd);
+}
+
 function meta::pure::functions::meta::findExpressionsForFunctionInValueSpecification(vs:ValueSpecification[1], funcs:Function<Any>[*]):FunctionExpression[*]
 {
-   $vs->match([fe:FunctionExpression[1] | let res = $fe.parametersValues->map(v | $v->findExpressionsForFunctionInValueSpecification($funcs));
-                                          if ( !($fe.func->instanceOf(Property) || $fe.func->instanceOf(QualifiedProperty)) && ($funcs->isEmpty() || $fe.func->in($funcs)), | [$fe]->concatenate($res), | $res );,
+   findExpressionsForFunctionInValueSpecification($vs, $funcs, [], []);
+}
+
+function <<access.private>> meta::pure::functions::meta::findExpressionsForFunctionInValueSpecification(vs:ValueSpecification[1], funcs:Function<Any>[*], visitedFuncs:FunctionDefinition<Any>[*], skipPackages:Package[*]):FunctionExpression[*]
```

Comment:
create a public method which takes shouldRecurse, but no visited for findExpressionsForFunctionInValueSpecification similar to how you did for findExpressionsForFunctionInFunctionDefinition